Pony/include/pony/Socket.hpp

```diff
@@ -6,36 +6,64 @@
 /* Original story "Networking for Game Programmers" by Glenn Fiedler
  */
 
-#ifndef PONY_SOCKET_HPP
-#define PONY_SOCKET_HPP
+#ifndef ___PONY__SOCKET_HPP___
+#define ___PONY__SOCKET_HPP___
 
 #include <pony/Address.hpp>
-#include <pony/SharedMacros.hpp>
+
+#if defined(_WIN32)
+    #include <winsock2.h>
+    #pragma comment( lib, "ws2_32.lib" )
+#endif
 
 namespace pony
 {
 
-class PONY_EXPORT Socket
+///HACK: to make Winsock happy we need in socketInit()/Kill()!
+inline bool socketsInit()
 {
+#if defined(_WIN32)
+    WSADATA WsaData;
+    return (WSAStartup(MAKEWORD(2,2), &WsaData) == 0);
+#else
+    return true;
+#endif
+}
+
+inline void socketsKill()
+{
+#if defined(_WIN32)
+    WSACleanup();
+#endif
+}
 
+/**
+ * @brief BSD Socket wrapper
+ */
+class Socket
+{
 public:
     Socket();
+    ~Socket();
 
     bool IsOpen() const;
 
-    bool Open(unsigned short port);
+    bool Open(const unsigned short port);
     void Close();
 
-    int Send(const Address & dst, const void * data, unsigned size);
+    int Send(const Address & dst, const void * const data, const unsigned size);
```

Comment:
`int32_t` or `int16_t` or anything that has a defined size would be better

Reply:
ok